File: src/dataset/Java/acronym_exercism-java-ThomasZumsteg/openAI/ChatGPT4_acronym_v4.java
public class Acronym {
    public static String generate(String phrase) {
        StringBuilder acronym = new StringBuilder();
        for (String word : phrase.split("\\W+")) {
            acronym.append(Character.toUpperCase(word.charAt(0)));
            for (int i = 1; i < word.length(); i++) {
                char c = word.charAt(i);
                if (Character.isUpperCase(c)) {
                    acronym.append(c);
                }
            }
        }
        return acronym.toString();
    }
}

File: src/dataset/typescript/acronym_Exercism-typescript-shybyte/openAI/ChatGPT4_acronym_v4.ts
const isUpperCase = (s: string) => s === s.toUpperCase();

const isLetter = (char: string) => /^[a-zA-Z]$/.test(char);

const wordToAcronym = (word: string) => {
    let acronym = word[0].toUpperCase();
    for (let i = 1; i < word.length; i++) {
        if (isUpperCase(word[i])) acronym += word[i];
    }
    return acronym;
};

const replacePunctuationWithSpace = (s: string) => s.replace(/[^a-zA-Z]/g, ' ');

const parse = (s: string) => {
    return replacePunctuationWithSpace(s)
        .trim()
        .split(/\s+/)
        .map(wordToAcronym)
        .join('');
};

export default { parse };